agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AgentMetricExporter.java

```diff
@@ -66,8 +74,15 @@ public CompletableResultCode export(Collection<MetricData> metrics) {
       logger.debug("exporter is not active");
       return CompletableResultCode.ofSuccess();
     }
-
     for (MetricData metricData : metrics) {
+      if (!Constant.NON_FILTERABLE_METRIC_NAMES.contains(metricData.getName())) {
+        for (MetricFilter metricFilter : metricFilters) {
+          if (!metricFilter.matches(metricData.getName())) {
+            // user configuration filtered out this metric name
+            return CompletableResultCode.ofSuccess();
+          }
+        }
+      }
```

Comment:
probably can extract this into common method in `Constant` class, e.g. `boolean shouldSkip(List<MetricFilter>)`, and share with TelemetryClient

Reply:
i moved it to MetricFilter as well.